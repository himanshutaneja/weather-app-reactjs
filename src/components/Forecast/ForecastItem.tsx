import { FC } from 'react';
import moment from 'moment';
import { ForecastDaily } from '../../store/types';
import { formatTemperature } from '../../helpers';
import {
  ForcastItemContainer,
  Day,
  TempMax,
  TempMin,
  WeatherIcon,
  Rain,
} from './styles';

interface ForecastItemProps {
  data: ForecastDaily;
}

const ForecastItem: FC<ForecastItemProps> = ({ data }) => {
  return (
    <ForcastItemContainer>
      <Day>{moment.unix(data.dt).format('ddd DD')}</Day>
      <TempMax>{formatTemperature(data.temp.max)}°</TempMax>
      <TempMin>{formatTemperature(data.temp.min)}°</TempMin>
      <WeatherIcon className={`wi wi-owm-${data.weather[0].id}`}></WeatherIcon>
      {data.rain && <Rain>{data.rain}%</Rain>}
    </ForcastItemContainer>
  );
};

export default ForecastItem;
