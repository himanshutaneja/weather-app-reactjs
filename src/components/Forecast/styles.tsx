import styled from 'styled-components';

export const ForcastContainer = styled.div`
  display: flex;
  padding: 20px;
  border-radius: 6px;
  box-shadow: 0 0 12px 0 rgb(0 0 0 / 20%);
  background-image: linear-gradient(#046f94, #3493ad);
  justify-content: space-evenly;
`;

export const ForcastItemContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const Day = styled.h3`
  font-size: 1.125rem;
  line-height: 1.66em;
  border-bottom: 1px solid;
`;

export const TempMax = styled.div`
  font-size: 2rem;
`;

export const TempMin = styled.div`
  font-size: 1.125rem;
`;

export const WeatherIcon = styled.i`
  font-size: 2.5rem;
  margin: 10px 0 15px 0;
`;

export const Rain = styled.div`
  font-size: 1.125rem;
`;
