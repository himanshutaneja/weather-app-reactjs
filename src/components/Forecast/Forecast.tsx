import { FC } from 'react';
import ForecastItem from './ForecastItem';
import { WeatherForecast } from '../../store/types';
import { ForcastContainer } from './styles';

interface ForecastProps {
  data: WeatherForecast;
}

const Forecast: FC<ForecastProps> = ({ data }) => (
  <ForcastContainer>
    {data.daily.slice(0, 5).map((dailyForecast) => (
      <ForecastItem key={dailyForecast.dt} data={dailyForecast} />
    ))}
  </ForcastContainer>
);

export default Forecast;
