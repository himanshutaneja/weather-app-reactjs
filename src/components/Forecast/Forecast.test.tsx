import { shallow, ShallowWrapper } from 'enzyme';
import Forecast from './Forecast';
import { weatherForecastMock } from '../../fixtures/WeatherForecast.fixture';

describe('Forecast', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<Forecast data={weatherForecastMock} />);
  });

  it('should render search bar correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
