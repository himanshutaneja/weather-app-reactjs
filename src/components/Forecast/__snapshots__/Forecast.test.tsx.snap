// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast should render search bar correctly 1`] = `
<styled.div>
  <ForecastItem
    data={
      Object {
        "dt": 1626660000,
        "rain": 11.1,
        "temp": Object {
          "day": 14.83,
          "eve": 12.81,
          "max": 15.26,
          "min": 8.14,
          "morn": 8.27,
          "night": 9.75,
        },
        "weather": Array [
          Object {
            "description": "broken clouds",
            "icon": "04d",
            "id": 803,
            "main": "Clouds",
          },
        ],
      }
    }
    key="1626660000"
  />
  <ForecastItem
    data={
      Object {
        "dt": 1626746400,
        "rain": 2,
        "temp": Object {
          "day": 16.09,
          "eve": 13.85,
          "max": 16.54,
          "min": 8.81,
          "morn": 9.07,
          "night": 11.31,
        },
        "weather": Array [
          Object {
            "description": "scattered clouds",
            "icon": "03d",
            "id": 802,
            "main": "Clouds",
          },
        ],
      }
    }
    key="1626746400"
  />
</styled.div>
`;
